airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/ctid/CtidGlobalStateManager.java

```diff
@@ -147,7 +146,7 @@ public CdcState getCdcState() {
   @Override
   public AirbyteStateMessage createFinalStateMessage(final AirbyteStreamNameNamespacePair pair, final JsonNode streamStateForIncrementalRun) {
     // Only incremental streams can be transformed into the next phase.
-    if (!resumableFullRefreshStreams.contains(pair)) {
+    if (!resumableFullRefreshStreams.contains(pair) && !nonResumableFullRefreshStreams.contains(pair)) {
```

Comment:
This condition looks very confusing, as their name suggests... it is like two contradicting conditions! 

is there a better function that can capture this?

Reply:
Wrapped this into `isIncrementalStream`